Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -149,10 +149,25 @@ class BrowserViewController: UIViewController {
         }
     }
 
-    private func finishEditingAndSubmit(url: NSURL) {
+    private func finishEditingAndSubmit(var url: NSURL) {
         urlBar.updateURL(url)
         urlBar.finishEditing()
-        tabManager.selectedTab?.loadRequest(NSURLRequest(URL: url))
+
+        if let tab = tabManager.selectedTab {
+            if ReaderMode.isReaderModeURL(url) {
+                if let readerMode = tab.getHelper(name: "ReaderMode") as? ReaderMode {
+                    // Switch to reader mode immediately when we detect it can be activated. The reader mode will still
+                    // call its delegate to let us know its state changed so that we can update the UI.
+                    readerMode.activateImmediately = true
+                    // We don't show the initial page when opening a reader: url. This will probably change to some overlay on top of the webview.
+                    tab.hideContent(animated: false)
```

Comment:
This isn't great since webViews are stacked, meaning going into reader mode actually shows the tab behind the current one. Sounds like you're planning on changing this, so this is fine for now, but please file a follow-up so this doesn't get forgotten.